src/components/InventoryV2/view/BatchDetailsModalForNursery.tsx

```diff
@@ -442,21 +440,28 @@ export default function BatchDetailsModal(props: BatchDetailsModalProps): JSX.El
                 </Grid>
 
                 <Grid item xs={12} padding={theme.spacing(1, 0, 1, 2)}>
-                  <Dropdown
-                    id='subLocationIds'
+                  <MultiSelect
+                    fullWidth={true}
                     label={strings.SUB_LOCATION}
-                    selectedValue={record.subLocationIds ? record.subLocationIds[0] : record.subLocationIds}
-                    options={availableSubLocations.map((subLocation) => ({
-                      label: subLocation.name,
-                      value: subLocation.id,
-                    }))}
-                    onChange={(subLocationId: string) =>
+                    onAdd={(subLocationId: string) => {
                       setRecord((previousValue) => ({
                         ...previousValue,
-                        subLocationIds: [Number(subLocationId)],
-                      }))
+                        subLocationIds: [...(previousValue?.subLocationIds || []), Number(subLocationId)],
+                      }));
+                    }}
+                    onRemove={(subLocationId: string) => {
+                      setRecord((previousValue) => ({
+                        ...previousValue,
+                        subLocationIds:
+                          previousValue?.subLocationIds?.filter((id: number) => `${id}` !== subLocationId) || [],
+                      }));
+                    }}
+                    options={
+                      new Map(availableSubLocations.map((subLocation) => [`${subLocation.id}`, subLocation.name]))
```

Comment:
the value can stay as a number, no need to string-ify it i think, multiselect component supports generics for key and value here